lib/core/src/examples/camera_navigation.rs

```diff
@@ -0,0 +1,90 @@
+use wasm_bindgen::prelude::*;
+
+use crate::display::render::css3d::Scene;
+use crate::display::render::css3d::Camera;
+use crate::display::render::css3d::html::HTMLObject;
+use crate::display::render::css3d::html::HTMLRenderer;
+use crate::system::web::StyleSetter;
+use crate::display::navigation::navigator::Navigator;
+
+use crate::animation::animator::continuous::ContinuousAnimator;
+use crate::animation::physics::inertia::*;
+
+use nalgebra::{Vector2, Vector3};
+use crate::set_stdout;
+use basegl_system_web::animation_frame_loop::AnimationFrameLoop;
+
+fn create_scene(dim:Vector2<f32>) -> Scene<HTMLObject> {
+    let mut scene : Scene<HTMLObject> = Scene::new();
+
+    let width  = dim.x / 2.0;
+    let height = dim.y / 2.0;
+
+    let positions = vec![
+        (0.5, 0.5),
+        (0.5, 1.5),
+        (1.5, 0.5),
+        (1.5, 1.5)
+    ];
+
+    let colors = vec![
+        (255, 0  , 0  ),
+        (0  , 255, 0  ),
+        (0  ,   0, 255),
+        (255, 255,   0)
+    ];
+
+    for i in 0..=3 {
+        let     object = HTMLObject::new("div");
+        let mut object = object.expect("Couldn't create div");
+        let (p_x, p_y) = positions[i];
+        object.set_dimensions(width, height);
+        object.set_position(Vector3::new(width * p_x, height * p_y, 0.0));
+        let (c_r, c_g, c_b) = colors[i];
+        let color = format!("rgb({}, {}, {})", c_r, c_g, c_b);
+        object.dom.set_property_or_panic("background-color", color);
+        scene.add(object);
+    }
+
+    scene
+}
+
+#[wasm_bindgen]
+#[allow(dead_code)]
+pub fn run_example_camera_navigation() {
+    set_stdout();
+    let renderer = HTMLRenderer::new("app").expect("Renderer couldn't be created");
+    renderer.container.dom.set_property_or_panic("background-color", "black");
+
+    let dimensions = renderer.dimensions();
+    let scene = create_scene(dimensions);
+
+    let mut camera  = Camera::perspective(45.0, dimensions.x / dimensions.y, 1.0, 1000.0);
+
+    let x = dimensions.x / 2.0;
+    let y = dimensions.y / 2.0;
+    let z = y * camera.get_y_scale();
+    camera.set_position(Vector3::new(x, y, z));
+
+    let mut event_loop = AnimationFrameLoop::new();
+
+    let zoom_speed   = 2.0;
+    let min_zoom     = 10.0;
+    let max_zoom     = 10000.0;
+    let camera_clone = camera.clone();
+    let navigator  = Navigator::new(
+        &mut event_loop,
+        &renderer.container,
+        camera_clone,
+        min_zoom,
+        max_zoom,
+        zoom_speed
+    );
+    let navigator  = navigator.expect("Couldn't create navigator");
+
```

Comment:
Few things which I believe we should change IN THE NEXT PR (lets talk about it when planning the sprint):

- Construction of that shouuld be simpler. I believe that our decision to pass here `event_loop` explicitly was bad. It should be encapsulated. Let's talk about it!
- Passing the container explicitly is cool, but we should have a mode where it just attaches the events to the whole window. Our application will have always 1 div and the whole window may want to listen for that. On the other hand, it is connected to the event system I am about to just start, so maybe lets leave it as is for now and wait for the event system to see how we can utilize it (I dont know how to do it yet) - this is just a note so you see what I'm thinking about.
- `min_zoom`, `max_zoom`, `zoom_speed` sohuld be configurable but not provided as part of the constructor - they should have default values when constructed. API (especially constructors) sohuld be simple. This is small point, if you feel you can change it now, lets do it.